crates/sncast/src/starknet_commands/verify.rs

```diff
@@ -0,0 +1,184 @@
+                        let file_content = std::fs::read_to_string(path)?;
+                        file_data.insert(
+                            relative_path.to_string_lossy().into_owned(),
+                            serde_json::Value::String(file_content),
+                        );
+                    }
+                }
+            }
+        }
+
+        // Serialize the JSON object to a JSON string
+        let source_code = serde_json::Value::Object(file_data);
+
+        // Create the JSON payload with "contract name," "address," and "source_code" fields
+        let payload = VerificationPayload {
+            contract_name: contract_name.to_string(),
+            contract_address: contract_address.to_string(),
+            source_code,
+        };
+
+        // Serialize the payload to a JSON string for the POST request
+        let json_payload = serde_json::to_string(&payload)?;
+
+        // Send the POST request to the explorer
+        let client = reqwest::Client::new();
+        let api_res = client
+            .post(self.gen_explorer_url()?)
+            .header("Content-Type", "application/json")
+            .body(json_payload)
+            .send()
+            .await
+            .context("Failed to send request to verifier API")?;
+
+        if api_res.status() == StatusCode::OK {
+            let message = api_res
+                .text()
+                .await
+                .context("Failed to read verifier API response")?;
+            Ok(VerifyResponse { message })
+        } else {
+            let message = api_res.text().await.context("Failed to verify contract")?;
+            Err(anyhow!(message))
+        }
+    }
+
+    fn gen_explorer_url(&self) -> Result<String> {
+        let api_base_url =
+            env::var("WALNUT_API_URL").unwrap_or_else(|_| "https://api.walnut.dev".to_string());
+        let path = match self.network.as_str() {
+            "mainnet" => "/v1/sn_main/verify",
+            "sepolia" => "/v1/sn_sepolia/verify",
+            _ => return Err(anyhow!("Unknown network")),
+        };
+        Ok(format!("{api_base_url}{path}"))
+    }
+}
+
+#[derive(Args)]
+#[command(about = "Verify a contract through a block explorer")]
+pub struct Verify {
+    /// Address of a contract to be verified
+    #[clap(short = 'a', long)]
+    pub contract_address: FieldElement,
+
+    /// Name of the contract that is being verified
+    #[clap(short, long)]
+    pub contract_name: String,
+
+    /// Block explorer to use for the verification
+    #[clap(short = 'v', long = "verifier", value_parser = ["walnut"], default_value = "walnut")]
+    pub verifier: String,
+
+    /// The network on which block explorer will do the verification
+    #[clap(short = 'n', long = "network", value_parser = ["mainnet", "sepolia"])]
+    pub network: String,
+
+    /// Assume "yes" as answer to confirmation prompt and run non-interactively
+    #[clap(long, default_value = "false")]
+    pub yes: bool,
```

Comment:
This option should be named differently, i think it's important to let the user know what he's confirming. There's only one prompt now, so we can name this precisely how we want, since we know which confirmation we want to skip here